web/shared/src/typeahead.ts

```diff
@@ -142,12 +144,25 @@ export function triage<T>(
             beginswithCaseSensitive.push(obj);
         } else if (lowerItem.startsWith(lowerQuery)) {
             beginswithCaseInsensitive.push(obj);
+        } else if (
+            lowerItem.includes(" " + lowerQuery) ||
+            lowerItem.includes("_" + lowerQuery) ||
+            lowerItem.includes("-" + lowerQuery) ||
+            lowerItem.includes("/" + lowerQuery)
+        ) {
+            // space, hyphen, dash and slash characters are considered word boundaries
```

Comment:
That's an interesting idea, and in fact this is how we detected word boundaries in another (much _much_ smaller) project I worked on earlier. However, this can wrongly consider diacritics word boundaries unless we remove them first... Thoughts? @alya @timabbott 

PS: In a bit of a rush as I write this so haven't yet given thought to how efficient it will be to check the character right before all occurrences of the query

Reply:
Hm, I don't find this easy to think through... Remind me, is this PR just for emoji typeahead, or others as well?

 If it's for emoji, looking at some examples of where the current algorithm vs. @evykassirer 's proposal differ would help.

Reply:
This PR affects all typeaheads; not just emojis